app/src/main/java/com/mr/nemo/dragonfly/ui/screen/auth/signup/SignUpScreen.kt

```diff
@@ -1,58 +1,163 @@
 package com.mr.nemo.dragonfly.ui.screen.auth.signup
 
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.foundation.ExperimentalFoundationApi
 import androidx.compose.foundation.clickable
 import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
 import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.requiredHeight
+import androidx.compose.foundation.pager.HorizontalPager
+import androidx.compose.foundation.pager.rememberPagerState
+import androidx.compose.material3.CenterAlignedTopAppBar
 import androidx.compose.material3.ExperimentalMaterial3Api
 import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.LinearProgressIndicator
 import androidx.compose.material3.Scaffold
-import androidx.compose.material3.TopAppBar
+import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.StrokeCap
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
 import com.mr.nemo.dragonfly.R
+import com.mr.nemo.dragonfly.domain.entity.auth.signup.SignUpPageContent
+import com.mr.nemo.dragonfly.ui.component.button.PrimaryButton
+import com.mr.nemo.dragonfly.ui.entitiy.signup.SignUpScreenEvent
+import com.mr.nemo.dragonfly.ui.entitiy.signup.SignUpScreenState
 import com.mr.nemo.dragonfly.ui.theme.DragonFlyTheme
+import org.koin.androidx.compose.koinViewModel
 
-@OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun SignUpScreen(
+    viewModel: SignUpViewModel = koinViewModel()
+) {
+    val state by viewModel.state.collectAsStateWithLifecycle()
+
+    SignUpScreen(
+        state = state,
+        onEvent = viewModel::onEvent
+    )
+}
+
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalFoundationApi::class)
+@Composable
+fun SignUpScreen(
+    state: SignUpScreenState,
+    onEvent: (SignUpScreenEvent) -> Unit,
     modifier: Modifier = Modifier
 ) {
     val spacing = DragonFlyTheme.spacing
     val typography = DragonFlyTheme.typography
     val colors = DragonFlyTheme.colors
+
+    val page = state.pages[state.currentPage]
+
     Scaffold(
+        modifier = modifier,
         topBar = {
```

Comment:
@bohdanzdaniekin I think you can extract this top bar into a separate component

Reply:
@slothdeboss fixed